rdwatch/scoring/tasks/animation_export.py

```diff
@@ -0,0 +1,698 @@
+)
+from rdwatch.scoring.models import (
+    AnimationModelRunExport,
+    AnimationSiteExport,
+    AnnotationProposalObservation,
+    AnnotationProposalSite,
+    EvaluationRun,
+    Observation,
+    Site,
+    SiteImage,
+)
+
+logger = logging.getLogger(__name__)
+
+label_mapping = {
+    'Active Construction': {
+        'color': (192, 0, 0),
+        'label': 'Active Construction',
+    },
+    'Post Construction': {
+        'color': (91, 155, 213),
+        'label': 'Post Construction',
+    },
+    'Site Preparation': {
+        'color': (255, 217, 102),
+        'label': 'Site Preparation',
+    },
+    'Unknown': {
+        'color': (112, 48, 160),
+        'label': 'Unknown',
+    },
+    'No Activity': {
+        'color': (166, 166, 166),
+        'label': 'No Activity',
+    },
+    'Positive Annotated': {
+        'color': (127, 0, 0),
+        'label': 'Positive Annotated',
+    },
+    'Positive Partial': {
+        'color': (0, 0, 139),
+        'label': 'Positive Partial',
+    },
+    'Positive Annotated Static': {
+        'color': (255, 140, 0),
+        'label': 'Positive Annotated Static',
+    },
+    'Positive Partial Static': {
+        'color': (255, 255, 0),
+        'label': 'Positive Partial Static',
+    },
+    'Positive Pending': {
+        'color': (30, 144, 255),
+        'label': 'Positive Pending',
+    },
+    'Positive Excluded': {
+        'color': (0, 255, 255),
+        'label': 'Positive Excluded',
+    },
+    'Negative': {
+        'color': (255, 0, 255),
+        'label': 'Negative',
+    },
+    'Ignore': {
+        'color': (0, 255, 0),
+        'label': 'Ignore',
+    },
+    'Transient Positive': {
+        'color': (255, 105, 180),
+        'label': 'Transient Positive',
+    },
+    'Transient Negative': {
+        'color': (255, 228, 196),
+        'label': 'Transient Negative',
+    },
+    'System Proposed': {
+        'color': (31, 119, 180),
+        'label': 'System Proposed',
+    },
+    'System Confirmed': {
+        'color': (31, 119, 180),
+        'label': 'System Confirmed',
+    },
+    'System Rejected': {
+        'color': (31, 119, 180),
+        'label': 'System Rejected',
+    },
+}
+
+
+def find_closest_observation(site_uuid, timestamp, model='Observation'):
+    if model == 'Observation':
+        observation_model = Observation
+    elif model == 'AnnotationProposalObservation':
+        observation_model = AnnotationProposalObservation
+    else:
+        raise ValueError(
+            "Invalid model. Choose either 'Observation' or 'AnnotationProposalObservation'."
+        )
+
+    # Fetch the closest observation where the date is before the SiteImage timestamp
+    closest_observation = (
+        observation_model.objects.filter(site_uuid=site_uuid, date__lte=timestamp)
+        .order_by('-date')
+        .first()
+    )
+
+    return closest_observation
+
+
+@shared_task
+def create_animation(
+    self, site_evaluation_id: UUID4, settings: GenerateAnimationSchema
```

Comment:
The `settings` object is not an instance of `GenerateAnimationSchema`, but a `GenerateAnimationSchema` that is serialized to a `dict`.

```suggestion
    self, site_evaluation_id: UUID4, settings: dict[str, Any]
```

Reply:
Yup celery was passing it as that.